### PR TITLE
Make bias optional in Dense and all Conv layers

### PR DESCRIPTION
**Pull request by sgugger**
This PR makes bias optional in the dense layer and all convolutional layers. This is necessary since most models (ResNets and variants) use convolutionla layers with no bias.

Since AD does not support optionals yet, this is implemented behind the scene using a boolean flag. That flag can be removed once AD deals properly with optionals, and we can keep the same inits.

### CODE REVIEW

File: Sources/TensorFlow/Layers/Convolutional.swift [hunk 1]
```diff
@@ -137,6 +142,8 @@ public struct Conv2D<Scalar: TensorFlowFloatingPoint>: Layer {
     @noDerivative public let padding: Padding
     /// The dilation factor for spatial dimensions.
     @noDerivative public let dilations: (Int, Int)
+    /// Workaround optionals not being handled by AD
```
Comment (dan-zheng) [inline Sources/TensorFlow/Layers/Convolutional.swift]: TF-1004 tracks enum differentiation support.
Could you please add references to TF-1004 in comments explaining that `useBias` is a workaround?
```suggestion
    // Note: `useBias` is a workaround for TF-1004: enum differentiation support.
```

File: Sources/TensorFlow/Layers/Convolutional.swift [hunk 2]
```diff
@@ -47,18 +49,19 @@ public struct Conv1D<Scalar: TensorFlowFloatingPoint>: Layer {
     ///   - dilation: The dilation factor for the temporal dimension.
     public init(
         filter: Tensor<Scalar>,
-        bias: Tensor<Scalar>,
+        bias: Tensor<Scalar>?,
```
Comment (dan-zheng) [inline Sources/TensorFlow/Layers/Convolutional.swift]: How about adding a default argument value  `nil` to all `bias` arguments?
```suggestion
        bias: Tensor<Scalar>? = nil,
```

File: Sources/TensorFlow/Layers/Convolutional.swift (hunk 1 above)
Comment (sgugger) [reply Sources/TensorFlow/Layers/Convolutional.swift]: Does enum differentiation imply optional differentiation? (Sorry if this is a dumb question, just want to make sure.)

File: Sources/TensorFlow/Layers/Convolutional.swift (hunk 2 above)
Comment (sgugger) [reply Sources/TensorFlow/Layers/Convolutional.swift]: Sure, I can add that on all inits.

File: Sources/TensorFlow/Layers/Convolutional.swift (hunk 1 above)
Comment (dan-zheng) [reply Sources/TensorFlow/Layers/Convolutional.swift]: Good point - enum differentiation support doesn't strictly imply `Optional` differentiation support.

I created a dedicated issue TF-1153 to track differentiation support for `Optional` operations, please reference that instead!

Comment (sgugger) [reply Sources/TensorFlow/Layers/Convolutional.swift]: Done, let me know if the phrasing is okay.

File: Sources/TensorFlow/Layers/Convolutional.swift [hunk 3]
```diff
@@ -104,14 +108,15 @@ public extension Conv1D where Scalar.RawSignificand: FixedWidthInteger {
         padding: Padding = .valid,
         dilation: Int = 1,
         activation: @escaping Activation = identity,
+        useBias: Bool = true,
```
Comment (Shashi456) [inline Sources/TensorFlow/Layers/Convolutional.swift]: Can a private variable be used in an initializer like this? I mean will we be able to set it to false externally?

Comment (sgugger) [reply Sources/TensorFlow/Layers/Convolutional.swift]: Here `useBias` is not the same as the internal flag, it's just to know if it has to pass a bias or nil to the main init.

File: Sources/TensorFlow/Layers/Convolutional.swift (hunk 1 above)
Comment (dan-zheng) [reply Sources/TensorFlow/Layers/Convolutional.swift]: Wording looks good!

**Pull request merged**